### PR TITLE
Move translations' namespace

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Based on #859

### CODE REVIEW

File: src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php [hunk 1]
```diff
@@ -15,7 +15,11 @@
 
 use Sylius\Resource\Model\TranslatableInterface;
 
-interface TranslatableEntityLocaleAssignerInterface
-{
-    public function assignLocale(TranslatableInterface $translatableEntity): void;
+interface_exists(\Sylius\Resource\Translation\TranslatableEntityLocaleAssignerInterface::class);
+
+if (false) {
+    interface TranslatableEntityLocaleAssignerInterface
```
Comment (diimpp) [inline src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php]: Shouldn't legacy `TranslatableEntityLocaleAssignerInterface` extend new `TranslatableEntityLocaleAssignerInterface` ?

File: src/Component/src/Translation/Provider/ImmutableTranslationLocaleProvider.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    /** @var array */
+    private $definedLocalesCodes;
+
+    private string $defaultLocaleCode;
+
+    public function __construct(array $definedLocalesCodes, string $defaultLocaleCode)
+    {
+        $this->definedLocalesCodes = $definedLocalesCodes;
+        $this->defaultLocaleCode = $defaultLocaleCode;
+    }
```
Comment (diimpp) [inline src/Component/src/Translation/Provider/ImmutableTranslationLocaleProvider.php]:
```suggestion
    public function __construct(
        private array $definedLocalesCodes,
        private string $defaultLocaleCode,
     ) {
    }
```

File: src/Component/src/Translation/Provider/TranslationLocaleProviderInterface.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Translation\Provider;
+
+interface TranslationLocaleProviderInterface
+{
+    /**
+     * @return string[]
+     */
```
Comment (diimpp) [inline src/Component/src/Translation/Provider/TranslationLocaleProviderInterface.php]:
```suggestion
    /** @return string[] */
```

File: src/Component/src/Translation/TranslatableEntityLocaleAssigner.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+use Sylius\Resource\Translation\Provider\TranslationLocaleProviderInterface;
+
+final class TranslatableEntityLocaleAssigner implements TranslatableEntityLocaleAssignerInterface
+{
+    private TranslationLocaleProviderInterface $translationLocaleProvider;
+
+    public function __construct(TranslationLocaleProviderInterface $translationLocaleProvider)
+    {
+        $this->translationLocaleProvider = $translationLocaleProvider;
+    }
```
Comment (diimpp) [inline src/Component/src/Translation/TranslatableEntityLocaleAssigner.php]:
```suggestion
    public function __construct(private TranslationLocaleProviderInterface $translationLocaleProvider)
    {
    }
```

File: tests/Application/src/Factory/BookFactory.php [hunk 5]
```diff
@@ -17,7 +17,7 @@
 use Sylius\Component\Resource\Factory\FactoryInterface;
 use Sylius\Component\Resource\Factory\TranslatableFactoryInterface;
 use Sylius\Component\Resource\Model\TranslatableInterface;
```
Comment (diimpp) [inline tests/Application/src/Factory/BookFactory.php]:
```suggestion
use Sylius\Resource\Factory\FactoryInterface;
use Sylius\Resource\Factory\TranslatableFactoryInterface;
use Sylius\Resource\Model\TranslatableInterface;
```

File: src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php (hunk 1 above)
Comment (loic425) [reply src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php]: Ouch you're right. Good catch

File: src/Component/src/Translation/Provider/ImmutableTranslationLocaleProvider.php (hunk 2 above)
Comment (loic425) [reply src/Component/src/Translation/Provider/ImmutableTranslationLocaleProvider.php]: Hum should not be a bc-break indeed cause definedLocalesCodes is private an is an array on contructor arguements.

File: tests/Application/src/Factory/LegacyBookFactory.php [hunk 6]
```diff
@@ -14,9 +14,9 @@
 namespace App\Factory;
 
 use Sylius\Component\Resource\Exception\UnexpectedTypeException;
-use Sylius\Component\Resource\Translation\Provider\TranslationLocaleProviderInterface;
 use Sylius\Resource\Factory\FactoryInterface;
 use Sylius\Resource\Model\TranslatableInterface;
+use Sylius\Resource\Translation\Provider\TranslationLocaleProviderInterface;
```
Comment (loic425) [inline tests/Application/src/Factory/LegacyBookFactory.php:19]: Maybe not a good idea here, cause it's a legacy factory.

File: src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php (hunk 1 above)
Comment (loic425) [reply src/Component/legacy/src/Translation/TranslatableEntityLocaleAssignerInterface.php]: fixed

File: tests/Application/src/Factory/LegacyBookFactory.php (hunk 6 above)
Comment (loic425) [reply tests/Application/src/Factory/LegacyBookFactory.php:19]: Hum there are some other changes to do on this legacy part on tests... I should make another PR to fix them all.

**Review by diimpp — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @loic425!

**Pull request merged**